### ISSUE TITLE
Terminal: step into brackets () needs double keystroke

### ISSUE THREAD

**Issue by fhgd** [jupyterlab/jupyterlab#16601]
Since JupyterLab 3.4.4 with Firefox (126 and 128) the Terminal has a problem with brackets (auto-close is off):

* type: '()' or '[]' or '{}'
* then you need to type 'cursor-left' twice in order to jump into the brackets

I expect only one keystroke for the '<-' cursor-left-key. In chrome this isnt an issue.

In jupyterlab < 3.4.4 this problem doesnt exists in firefox. Maybe this is come from xtermjs... Does anyone has an idea?

**Comment by krassowski** [comment]
JupyterLab 3.x is no longer receiving bug fixes, only security updates (see README, [blog post](https://blog.jupyter.org/jupyterlab-3-end-of-maintenance-879778927db2), [documentation](https://jupyterlab.readthedocs.io/en/latest/getting_started/lifecycle.html)).

Is this reproducible for you in JupyterLab 4.x?

**Comment by fhgd** [comment]
Yes, this bug appears in current 4.2.4 and seems to be introduced in 3.4.4.

**Comment by JasonWeill** [comment]
@fhgd Thank you for your contribution! I'm not seeing this behavior in Firefox 127 with the newest (alpha) version of JupyterLab. Are you running any browser extensions or JupyterLab extensions that may be intercepting user input? If so, can you still reproduce this using a browser without these extensions or with JupyterLab with these extensions disabled? Also, which operating system and which keyboard layout (US, UK, etc) are you using? Thanks!

**Comment by JasonWeill** [comment]
@fhgd Any updates regarding the questions above? Thanks!

**Comment by fhgd** [comment]
Yes.I discovered that this issue is only with xterm and my alternative keyboard layout: Ubuntu 22.04: Deutsch, Neo 2. (Neo has all brackets in layer 3 and additional cursor-key on layer 4. This is very convenient for programming ...).

This means there is no issue with:
- JupyterLab-XTerm and normal cursor-keys
- (Ipython) Console and Notebook with Neo, Layer 3 and 4
- JupyterLab-XTerm (< 3.4.4) with Neo, Layer 3 and 4
